lib/ecto/model/optimistic_locking.ex

```diff
@@ -0,0 +1,18 @@
+defmodule Ecto.Model.OptimisticLocking do
+  defmacro optimistic_locking(field) do
```

Comment:
English question: should this be called `optimistic_locking` or `optimistic_lock`? It feels we should call it the latter because we are passing the name of the field, so the field is effectively the `optimistic_lock`. It is also a bit shorter. :D

If we change the macro name, we should also change the model one.